src/fastcdc.rs

```diff
@@ -0,0 +1,281 @@
+use super::Engine;
+use std::default::Default;
+use std::cmp;
+use std::mem;
+use Gear;
+
+pub struct FastCDC {
+    current_chunk_size: u64,
+    gear: Gear,
+}
+
+impl Default for FastCDC {
+    fn default() -> Self {
+        FastCDC {
+            current_chunk_size: 0,
+            gear: Gear::default(),
+        }
+    }
+}
+
+
+impl Engine for FastCDC {
+    type Digest = u64;
```

Comment:
Interesting point, I had only been looking at rolling checksums (hence the crate name) and hadn't really thought about non-checksum based alternatives for doing chunking.